apps/aetx/src/aetx.erl

```diff
@@ -209,6 +210,12 @@ gas_price(#aetx{ type = Type, cb = CB, tx = Tx }) when ?IS_CONTRACT_TX(Type) ->
 gas_price(#aetx{}) ->
     undefined.
 
+-spec min_gas_price(Tx :: tx(), Height :: aec_blocks:height()) -> MinGasPrice :: non_neg_integer().
+min_gas_price(AETx = #aetx{ type = Type, cb = CB, tx = Tx }, Height) when ?IS_CONTRACT_TX(Type) ->
+    min(CB:gas_price(Tx), 1 + (CB:fee(Tx) div gas_limit(AETx, Height)));
```

Comment:
```suggestion
    %% There are two limit cases: zero gas used for contract execution; max gas used for contract execution.
    min( CB:fee(Tx) div min_gas(AETx)
       , (CB:fee(Tx) + (CB:gas(Tx) * CB:gas_price(Tx))) div gas_limit(AETx, Height)
       );
```

Reply:
No, I don't agree... I viewed the implicit gas price of the Fee as one gas price, and the dynamic gas price as a second gas price, simply returning the smallest of the two... One could try to compute some sort of average gas price like you suggest but that is probably harder to reason about for the miner?

Reply:
Agreed, easier to reason about.

Also the minimum computatio may be more complex:
```
   %% Both the miner reward and the tx gas increase linearly with the gas used:
    %% * Miner reward: `CB:fee(Tx) + (<gas used> * CB:gas_price(Tx))`.
    %% * Tx gas: `min_gas(AETx) + <gas used>`.
    %%
    %% What is the minimum of the ratio between the miner reward and the tx gas?
```